src/replay_stage.rs

```diff
@@ -160,14 +148,11 @@ impl ReplayStage {
 
                 // TODO: Remove this soon once we boot the leader from ClusterInfo
                 if scheduled_leader != current_leader {
-                    did_rotate = true;
                     cluster_info.write().unwrap().set_leader(scheduled_leader);
-                }
-
-                if !already_leader && my_id == scheduled_leader && did_rotate {
```

Comment:
The point of the conditions that were here previously was to truncate the number of entries written when this node transitioned from validator to leader. The way you have it now, if any leader rotation occurs between two nodes, it'll truncate entries. Is that right ? It sounds right to me but it is different from what we used to do for a while.

Reply:
Yup! Now we process_entries() one slot at at time, so it precludes that possibility